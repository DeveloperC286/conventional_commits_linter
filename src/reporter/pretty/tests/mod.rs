use rstest::rstest;

use super::*;

#[rstest(
    commit_message,
    snapshot_name,
    case(
        "feat()!: yargs-parser now throws on invalid combinations of config\n\n",
        "test_pretty_print_case_1"
    ),
    case("Release 1.0.126", "test_pretty_print_case_2"),
    case(
        "Merge branch 'fix-ocsp-signer-check' into 'master'",
        "test_pretty_print_case_3"
    )
)]
fn test_pretty_print(commit_message: &str, snapshot_name: &str) {
    // Given
    let linting_errors = crate::linter::lint_commit_message(commit_message, false);

    // When
    let reporting = print(None, commit_message, &linting_errors);

    // Then
    insta::assert_snapshot!(snapshot_name, reporting);
}

#[rstest(
    commit_messages,
    snapshot_name,
    case(
        &["feat()!: yargs-parser now throws on invalid combinations of config\n\n", "doc(webpack):webpack example (#1436)"],
        "test_pretty_print_all_case_1"
    ),
    case(
        &["refactor(ts support)!: ship yargs.d.ts (#1671)", "feat!: drop support for EOL Node 8 (#1686)"],
        "test_pretty_print_all_case_2"
    ),
)]
fn test_pretty_print_all(commit_messages: &[&str], snapshot_name: &str) {
    // Given
    let commits: Vec<Commit> = commit_messages
        .iter()
        .enumerate()
        .map(|(index, commit_message)| Commit {
            oid: git2::Oid::from_str(&*format!("{}", index)).unwrap(),
            message: commit_message.to_string(),
        })
        .collect();
    let linting_errors = crate::linter::lint_commits(&commits, false);

    // When
    let reporting = print_all(&commits, &linting_errors);

    // Then
    insta::assert_snapshot!(snapshot_name, reporting);
}
