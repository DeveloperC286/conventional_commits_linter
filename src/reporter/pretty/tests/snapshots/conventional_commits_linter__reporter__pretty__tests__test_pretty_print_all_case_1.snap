---
source: src/reporter/pretty/tests/mod.rs
expression: reporting

---
[1;31mCommit Hash[0m - 0000000000000000000000000000000000000000
[1;31mMessage[0m - "feat()!: yargs-parser now throws on invalid combinations of config\n\n"
	[1;31mX[0m - Commit title does not comply with the Conventional Commits V1.0.0 specification.
	[1;31mX[0m - Commit title has a scope which is empty.

[1;31mCommit Hash[0m - 1000000000000000000000000000000000000000
[1;31mMessage[0m - "doc(webpack):webpack example (#1436)"
	[1;31mX[0m - Commit title does not comply with the Conventional Commits V1.0.0 specification.
	[1;31mX[0m - Commit title has no space after the colon preceding the Conventional Commits type and scope.

[1;31mX[0m - Found 4 separate linting errors across 2 commits.
